src/main/java/com/splunk/logging/HttpEventCollectorLogbackAppender.java

```diff
@@ -97,21 +99,52 @@ public void stop() {
     }
 
     @Override
-    protected void append(ILoggingEvent event) {
+    protected void append(E e) {
+        if (e instanceof ILoggingEvent) {
+            sendEvent((ILoggingEvent) e);
+        } else {
+            sendEvent(e);
+        }
+    }
+
+    private void sendEvent(ILoggingEvent event) {
         event.prepareForDeferredProcessing();
-        event.getCallerData();
+        if (event.hasCallerData()) {
+            event.getCallerData();
+        }
+
         MarkerConverter c = new MarkerConverter();
         if (event != null && started) {
             this.sender.send(
                     event.getLevel().toString(),
-                    _layout.doLayout(event),
+                    _layout.doLayout((E) event),
                     _includeLoggerName ? event.getLoggerName() : null,
                     _includeThreadName ? event.getThreadName() : null,
                     _includeMDC ? event.getMDCPropertyMap() : null,
                     (!_includeException || event.getThrowableProxy() == null) ? null : event.getThrowableProxy().getMessage(),
                     c.convert(event)
-                    );
+            );
+        }
+    }
+
+    // send non ILoggingEvent such as ch.qos.logback.access.spi.IAccessEvent
+    private void sendEvent(E e) {
+        String message = _layout.doLayout(e);
+        if (message == null) {
+            throw new IllegalArgumentException(String.format(
+                    "The logback layout %s is probably incorrect, " +
+                            "and fails to format the message.",
+                    _layout.toString()));
         }
+        this.sender.send(
```

Comment:
I agree this looks odd. But currently the `HttpEventCollectorSender.send` method asks for information like logging event level and exception, which are not included in an access event. If you look at the code above from line 121~124, they are similar with the code here when the `_includeLoggerName`/`_includeThreadName`/`_includeMDC`/`_includeException` are set to false.

Probably we need to refine the interface for `HttpEventCollectorSender.send` to address this issue later on, but since this is my first PR for this project, I don't want to change too much to break anything accidentally.